charms/knative-operator/src/charm.py

```diff
@@ -122,7 +101,15 @@ def _main(self, event):
         """Event handler for changing Pebble configuration and applying k8s resources."""
         # Update Pebble configuration layer if it has changed
         self.unit.status = MaintenanceStatus("Configuring Pebble layer")
-        self._update_layer(event)
+        try:
+            update_layer(self._app_name, self._container, self._knative_operator_layer, logger)
+        except ErrorWithStatus as e:
+            self.model.unit.status = e.status
+            if isinstance(e.status, BlockedStatus):
+                logger.error(str(e.msg))
+            else:
+                logger.info(str(e.msg))
```

Comment:
Let's better use `logger.error` if we are showing the message from an exception.

Reply:
even if coming from a maintenance status? I chose `info` because the charm would resolve in the future, versus `BlockedStatus` requires human intervention (similar to an error in script)